### DISCUSSION TITLE
BRouter translations

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#382] [Q&A] [answered]
BRouter texts are only in English. It would be nice if we translate them in other languages.

(there are also some texts inside the code, which are currently difficult to translate)

I will also send the translations to BRouter, if they want to include them in BRouter app too.

- The 1st set of strings is here: [strings.xml.txt](https://github.com/devemux86/cruiser/files/13672932/strings.xml.txt)

- The 2nd set of strings is below: https://github.com/devemux86/cruiser/discussions/382#discussioncomment-7819945

- The last set of string is below: https://github.com/devemux86/cruiser/discussions/382#discussioncomment-7852874

cc @afischerdev

---

- [x] English
- [x] العربية
- [x] Català
- [x] Nederlands
- [x] Français
- [x] Deutsch
- [x] Ελληνικά
- [x] Italiano
- [x] 한국인
- [x] Polski
- [x] Español

**Comment by Ciutnik** [comment]
I am pleased to post the Polish translation.
[strings_pl.xml.txt](https://github.com/devemux86/cruiser/files/13401383/strings_pl.xml.txt)

I'm not sure if the last line should be translated:
`<string name="channel_name">Downloads</string>`

If so, it should be:
`<string name="channel_name">Pobrania</string>`

**Reply by devemux86** [reply]
@Ciutnik thanks!

Indeed, the following strings do not need translation:
- app_name
- profile_filename_example
- notification_channel_id
- channel_name

**Reply by Ciutnik** [reply]
Alternatively, although it's probably not necessary, you can change it to make the translation 100%: 
`"profile_filename_example">filename.brf` to `nazwapliku.brf` 🙂

**Comment by devemux86** [comment]
I finished the other translations: [brouter_translations.zip](https://github.com/devemux86/cruiser/files/13403382/brouter_translations.zip)

You can check your language (compared to English) to see if it needs improvements.

**Comment by airald** [comment]
I will translate Catalan.

[strings_ca.xml.txt](https://github.com/devemux86/cruiser/files/13413736/strings_ca.xml.txt)

**Reply by airald** [reply]
updated the translation with some typing error

[strings_ca.xml.txt](https://github.com/devemux86/cruiser/files/13647639/strings_ca.xml.txt)

**Comment by bios4** [comment]
Minor update for the German strings attached...
[strings_de.xml.txt](https://github.com/devemux86/cruiser/files/13413790/strings_de.xml.txt)

**Comment by abdullahO2** [comment]
update for the Arabic strings:
[strings_ar.xml.txt](https://github.com/devemux86/cruiser/files/13415392/strings_ar.xml.txt)

**Comment by devemux86** [comment]
Thanks everyone for your contributions! :+1: 

I also posted the translations in the BRouter repository:
- https://github.com/abrensch/brouter/pull/645

We may include more texts in strings.xml in the future, so we can translate them as well.

**Comment by devemux86** [comment]
We have the latest strings to translate in BRouter. :slightly_smiling_face: 

You can find the English file here: [strings.xml.txt](https://github.com/devemux86/cruiser/files/13636099/strings.xml.txt)

@afischerdev thanks for separating the strings.

**Reply by Ciutnik** [reply]
I am posting the Polish translation here. I hope it reaches the BRouter developers 🙂 By the way, thank you for this, powerful tool! 👍
[🇵🇱 strings-pl.xml.txt](https://github.com/devemux86/cruiser/files/13640799/strings-pl.xml.txt)

**Reply by airald** [reply]
Here the Catalan files. Thank's fot the effort.
[strings_ca.xml.txt](https://github.com/devemux86/cruiser/files/13644463/strings_ca.xml.txt)

**Reply by bios4** [reply]
German translation attached...

Some terms are very hard to translate properly/in a correct form without the corresponding context, so the BRouter **developer** needs to **double-check** it (he's German native speaker I suppose)!

[strings_de.xml.txt](https://github.com/devemux86/cruiser/files/13644969/strings_de.xml.txt)

**Reply by devemux86** [reply]
Thanks everyone! :+1: 

@bios4 in German translation the texts at lines 50-54 in `msg_status_result` need more translation?

**Reply by bios4** [reply]
> @bios4 in German translation the texts at lines 50-54 in `msg_status_result` need more translation?

Eventually, yes.

Unfortunately I don't know the context (when is this displayed; what does "mem" mean; ...), so I was not able to translate this.
Also I don't know if the upper/lowercase string beginnings are correct in all of these "msg_"-variables.

Therefore my hint that it should be double checked by the BRouter developer.

**Reply by devemux86** [reply]
"mem" should be memory, the others are full words so they can be translated.
If we export a gpx in [BRouter website](https://brouter.de/brouter-web/) this info appears as comment at the top.

About the case, if a language allows lowercase, we can follow the English case.

**Reply by bios4** [reply]
> If we export a gpx in [BRouter website](https://brouter.de/brouter-web/) this info appears as comment at the top.

OK, nice to know...

But it seems that there are a few additional strings, which are different/not available in the strings.xml.txt file ("track-length", "cost", "energy" is missing; "plain-ascend" != "plain ascend", "time" != "estimated time", ...):
![grafik](https://github.com/devemux86/cruiser/assets/145460147/75bc37dd-c4bb-447c-855e-6e2014b38fd3)

> if a language allows lowercase, we can follow the English case.

This is not true in really all cases, as it depends e.g. if the message is shown as a standalone text (then it should start with uppercase letters) or somehow part of a "combined message" with any prefix (then it could begin with lowercase letters). Also I don't know if the English strings are correct in all cases - e.g. why is "msg_wpt_selection" starting with lowercase letters, "msg_no_wpt_selection" with uppercase?

So it should just be checked by someone who nows all occurences and situations where the strings are shown.

**Reply by devemux86** [reply]
> there are a few additional strings

The strings we translate are for the BRouter app.
The website may use more or different strings.

Important is to translate all strings so we complete all languages.
Improvements may be done later based on the use. :slightly_smiling_face: 

e.g. not all strings are seen if we only use the BRouter downloader.

**Reply by abdullahO2** [reply]
Here are the Arabic translation strings, and thank you for these wonderful developments:
[strings_ar.xml.txt](https://github.com/devemux86/cruiser/files/13647246/strings_ar.xml.txt)

**Comment by airald** [comment]
Also spanish translation

[strings_sp.xml.txt](https://github.com/devemux86/cruiser/files/13647684/strings_sp.xml.txt)

**Comment by devemux86** [comment]
@0709wiwiwi provided the Nederlands translation. :+1: 

[strings_nl.xml.txt](https://github.com/devemux86/cruiser/files/13661633/strings_nl.xml.txt)

We still have the Français, Ελληνικά, Italiano languages,
but I can translate them. :slightly_smiling_face:

**Comment by devemux86** [comment]
All translations exist in my BRouter repository: https://github.com/devemux86/brouter/commit/566df97a64b3429a54f35e71113901a2927121c3

I will send them to BRouter soon, if you'd like to do some final checking.

**Reply by umpfmpf** [reply]
What do you mean with "NoGo Selection" in German?
Is it like Avoidance?

**Reply by devemux86** [reply]
NoGo is a BRouter term for blocked areas. See also the [BRouter website](https://brouter.de/brouter-web/) tools at the left.
(Not sure we need to translate such terms)

**Reply by umpfmpf** [reply]
If we translate, then perhaps "Vermeidungen"

**Reply by bios4** [reply]
Specific terms should NOT be translated from my POV, so I suggest to keep "NoGo" as is also for German.

Instead of that, I think with "plain ascend" a flat ascend is meant, and not an easy/simple one!?
So I suggest to change "schlichter Aufstieg" to "flacher Aufstieg" @devemux86.

**Comment by devemux86** [comment]
And the last set of strings for translation:  [strings.xml.txt](https://github.com/devemux86/cruiser/files/13674255/strings.xml.txt)

cc @0709wiwiwi @abdullahO2 @airald @bios4 @Ciutnik

**Reply by Ciutnik** [reply]
Polish done! I added a number (2) to the name so that it would not be confused with the previous set.
[strings.xml-pl(2).txt](https://github.com/devemux86/cruiser/files/13673576/strings.xml-pl.2.txt)

**Reply by airald** [reply]
Catalan and spanish too.
[strings_sp-2.xml.txt](https://github.com/devemux86/cruiser/files/13673741/strings_sp-2.xml.txt)
[strings_ca-2.xml.txt](https://github.com/devemux86/cruiser/files/13673742/strings_ca-2.xml.txt)

**Reply by umpfmpf** [reply]
German too.
[strings.xml-de(2).txt](https://github.com/devemux86/cruiser/files/13673864/strings.xml-de.2.txt)

**Reply by bios4** [reply]
Suggestion of two minor changes for the German ^^ translations:

`<string name="summary_new_version">Die neue Datenversion verlangt ein BRouter-Update. Bitte aktualisieren</string>`
`<string name="msg_download_started">"Download läuft..."</string>` (why is this string the only one wrapped in quotation marks?)

**Reply by devemux86** [reply]
> why is this string the only one wrapped in quotation marks?

Indeed, I will fix it in all translations.

**Reply by abdullahO2** [reply]
Arabic translation
[strings_ar.xml.txt](https://github.com/devemux86/cruiser/files/13674288/strings_ar.xml.txt)

**Reply by devemux86** [reply]
@0709wiwiwi provided the Nederlands translation. 👍

[strings_nl.xml.txt](https://github.com/devemux86/cruiser/files/13675026/strings_nl.xml.txt)

I will do the other translations and send them to BRouter.

**Reply by devemux86** [reply]
All new translations are in my BRouter repository: https://github.com/devemux86/brouter/commit/37c6fc0d5c78a09d6a107ca7199f11ca73fa58c1

I will send them to BRouter.

**Comment by devemux86** [comment]
All BRouter translations are now available for testing in the latest:
- https://github.com/devemux86/cruiser/discussions/390

**Comment by devemux86** [comment, marked as answer]
BRouter translations in all Cruiser languages is available in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Thanks everyone for the translations! :+1: 

---

The translations will also be available in the next BRouter app.

**Comment by devemux86** [comment]
The [pull request](https://github.com/abrensch/brouter/pull/650) with BRouter's final translated texts was merged into its repository.
Next BRouter **app** will be available in 11 languages. Τhanks again everyone for your help! :+1: 

---

**Internal** BRouter in Cruiser is already available in all supported Cruiser languages.
